app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BaseRecyclerArrayAdapter.java

```diff
@@ -24,14 +24,17 @@
 public abstract class BaseRecyclerArrayAdapter<VIEW_HOLDER extends BaseRecyclerViewHolder<DATA>, DATA extends IdentifiedData>
         extends RecyclerView.Adapter<VIEW_HOLDER> {
 
-    private final OnItemClickListener<DATA> mListener;
+    private final OnItemClickListener<DATA> mOnItemClickListener;
+    private final OnNewDataListener<DATA> mOnNewDataListener;
     private final SortedList<DATA> mSortedList;
     private final Map<String, DATA> mDataById;
 
     public BaseRecyclerArrayAdapter(Class<DATA> klass, final Comparator<DATA> dataComparator,
-                                    OnItemClickListener<DATA> listener) {
+                                    OnItemClickListener<DATA> onItemClickListener,
+                                    OnNewDataListener<DATA> onNewDataListener) {
         mSortedList = new SortedList<>(klass, new SortedListCallback<>(dataComparator));
-        mListener = listener;
+        mOnItemClickListener = onItemClickListener;
+        mOnNewDataListener = onNewDataListener;
```

Comment:
Unlike the OnItemClick listener, I dunno if that listener should be mandatory (constructor parameter)
If it would be an optional listener (that is registered through a set method) if wouldn't have any effect on the Sensors-files for example.